blakserv/ccode.c

```
@@ -2344,3 +2351,76 @@ blak_int C_MinigameStringToNumber(int object_id,local_var_type *local_vars,
 	
 	return ret_val.int_val;
 }
+
+// --- BEGIN MULTI-PIPE WEBHOOK SUPPORT ---
+#ifdef BLAK_PLATFORM_WINDOWS
+#include <time.h>
+#define NUM_WEBHOOK_PIPES 10
+
+void SendWebhookRoundRobin(const char* message, int len) {
+    static int last_pipe_index = 0;
+    char full_message[1024];
+    time_t now = time(NULL);
+    int msg_len = snprintf(full_message, sizeof(full_message),
+         "{\"timestamp\":%ld,\"message\":\"%.*s\"}", (long)now, len, message);
+    if (msg_len < 0 || msg_len >= (int)sizeof(full_message)) {
+		// If snprintf fails or the message is too long, truncate it
+        msg_len = (int)sizeof(full_message) - 1;
+        full_message[msg_len] = '\0';
+    }
+    for (int i = 0; i < NUM_WEBHOOK_PIPES; ++i) {
+        int pipe_index = (last_pipe_index + i) % NUM_WEBHOOK_PIPES;
+        char pipe_name[64];
+        snprintf(pipe_name, sizeof(pipe_name), "\\\\.\\pipe\\m59apiwebhook%d", pipe_index + 1);
+        HANDLE hPipe = CreateFileA(pipe_name, GENERIC_WRITE, 0, NULL, OPEN_EXISTING, 0, NULL);
+        if (hPipe != INVALID_HANDLE_VALUE) {
+            DWORD bytesWritten;
+            WriteFile(hPipe, full_message, (DWORD)msg_len, &bytesWritten, NULL);
+            CloseHandle(hPipe);
+            last_pipe_index = (pipe_index + 1) % NUM_WEBHOOK_PIPES;
+            return;
+        }
+    }
+
+}
+#else
+#include <time.h>
+#define NUM_WEBHOOK_PIPES 1
+void SendWebhookRoundRobin(const char* message, int len) {
+    // For Linux use a single pipe
+    char full_message[1024];
+    time_t now = time(NULL);
+	
+    int msg_len = snprintf(full_message, sizeof(full_message),
+         "{\"timestamp\":%ld,\"message\":\"%.*s\"}", (long)now, len, message);
+    if (msg_len < 0 || msg_len >= (int)sizeof(full_message)) {
+        msg_len = (int)sizeof(full_message) - 1;
+        full_message[msg_len] = '\0';
+    }
+    int fd = open("/tmp/m59apiwebhook1", O_WRONLY | O_NONBLOCK);

```

This will silently fail if more than one server is run on the same machine.